### PR TITLE
Instrumentation Oracle.MDA - traces

### LINKED ISSUE #3192: README update: Add Oracle .NET OpenTelemetry provider
Oracle recently released its .NET OpenTelemetry provider that works with both Oracle.ManagedDataAccess.Core (ODP.NET Core) and Oracle.ManagedDataAccess (managed ODP.NET). Can you update the [Instrumentation Libraries README](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/internal/instrumentation-libraries.md) Databases table with the following or something similar?
Library | Tracing Support | Metrics Support | Databases Tested | Notes
-- | -- | -- | -- | --
| [Oracle.ManagedDataAccess.Core](https://www.nuget.org/packages/Oracle.ManagedDataAccess.Core) | Yes |  | [Oracle Database](https://www.oracle.com/database/)  |  Requires [Oracle.ManagedDataAccess.OpenTelemetry](https://www.nuget.org/packages/Oracle.ManagedDataAccess.OpenTelemetry) NuGet package |
| [Oracle.ManagedDataAccess](https://www.nuget.org/packages/Oracle.ManagedDataAccess) | Yes |  | [Oracle Database](https://www.oracle.com/database/) |  Requires [Oracle.ManagedDataAccess.OpenTelemetry](https://www.nuget.org/packages/Oracle.ManagedDataAccess.OpenTelemetry) NuGet package |Library	[Tracing Support](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/config.md#instrumented-traces-libraries-and-frameworks)	[Metrics Support](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/config.md#instrumented-metrics-libraries-and-frameworks)

I realize I could file a PR with the README change. However, going through the EasyCLA from a corporate standpoint is a big unknown in terms of time commitment and effort to complete when the change is straightforward and the update is from public info.

Thanks!

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Fixes #3192

## What

Support for [Oracle.ManagedDataAccess.Core](https://www.nuget.org/packages/Oracle.ManagedDataAccess.Core) and [Oracle.ManagedDataAccess](https://www.nuget.org/packages/Oracle.ManagedDataAccess) traces instrumentation from 23.4.0 together with support for   `OTEL_DOTNET_AUTO_ORACLEMDA_SET_DBSTATEMENT_FOR_TEXT` environment variable.

## Tests

CI + executed new tests on Windows with docker subsytem

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] Documentation is updated.
- [x] New features are covered by tests.

## Notes

~~I will keep the PR as a draft as long as Oracle.ManagedDataAccess.Core and Oracle.ManagedDataAccess will be not released as stable packages.~~

### CODE REVIEW

**Comment by Kielek** [comment]
Ready for review. We are waiting only for the stable version of the library.

Minimal changes, mainly for documentation part are expected.

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,11 @@ This component adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.h
   and upcoming End of Support date for .NET version.
 - Experimental support for ARM64 on Ubuntu, Alpine and Debian based images.
 - Experimental ARM64 support for the `OpenTelemetry.AutoInstrumentation` NuGet package.
+- Support for [Oracle.ManagedDataAccess.Core](https://www.nuget.org/packages/Oracle.ManagedDataAccess.Core)
+  and [Oracle.ManagedDataAccess](https://www.nuget.org/packages/Oracle.ManagedDataAccess)
+  traces instrumentation from 23.3.2-dev together with support for
+  `OTEL_DOTNET_AUTO_ORACLEMDA_SET_DBSTATEMENT_FOR_TEXT` environment variable.
+  ARM64 platform is not supported.
```
Comment (pjanotti) [inline CHANGELOG.md:38]: Q.: why ARM64 is not supported? Is it a limitation of the Oracle package?

File: test/IntegrationTests/OracleCollection.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        {
+            Port = TcpPortProvider.GetOpenPort();
+        }
+    }
+
+    public bool IsCurrentArchitectureSupported { get; } = EnvironmentTools.IsX64();
+
+    public int Port { get; }
+
+    public string Password { get; } = $"@{Guid.NewGuid():N}";
```
Comment (pjanotti) [inline test/IntegrationTests/OracleCollection.cs:36]: 👍🏼

**Review by pjanotti — APPROVED** [review]
LGTM, one Q, just curious why...

**Review by rajkumar-rangaraj — APPROVED** [review]

**Comment by Kielek** [comment]
Removing `do NOT merge` label. Stable version is out.

**Review by lachmatt — APPROVED** [review]

**Pull request merged**